contract/models/contract.py

```diff
@@ -0,0 +1,509 @@
+                or self.partner_id.with_context(
+                    force_company=self.company_id.id,
+                ).property_product_pricelist.currency_id
+            )
+        return currency or self.journal_id.currency_id or self.company_id.currency_id
+
+    @api.depends(
+        "manual_currency_id", "pricelist_id", "partner_id", "journal_id", "company_id",
+    )
+    def _compute_currency_id(self):
+        for rec in self:
+            if rec.manual_currency_id:
+                rec.currency_id = rec.manual_currency_id
+            else:
+                rec.currency_id = rec._get_computed_currency()
+
+    def _inverse_currency_id(self):
+        """If the currency is different from the computed one, then save it
+        in the manual field.
+        """
+        for rec in self:
+            if rec._get_computed_currency() != rec.currency_id:
+                rec.manual_currency_id = rec.currency_id
+            else:
+                rec.manual_currency_id = False
+
+    def _compute_invoice_count(self):
+        for rec in self:
+            rec.invoice_count = len(rec._get_related_invoices())
+
+    def action_show_invoices(self):
+        self.ensure_one()
+        tree_view = self.env.ref("account.view_invoice_tree", raise_if_not_found=False)
+        form_view = self.env.ref("account.view_move_form", raise_if_not_found=False)
+        action = {
+            "type": "ir.actions.act_window",
+            "name": "Invoices",
+            "res_model": "account.move",
+            "view_mode": "tree,kanban,form,calendar,pivot,graph,activity",
+            "domain": [("id", "in", self._get_related_invoices().ids)],
+        }
+        if tree_view and form_view:
+            action["views"] = [(tree_view.id, "tree"), (form_view.id, "form")]
+        return action
+
+    @api.depends("contract_line_ids.date_end")
+    def _compute_date_end(self):
+        for contract in self:
+            contract.date_end = False
+            date_end = contract.contract_line_ids.mapped("date_end")
+            if date_end and all(date_end):
+                contract.date_end = max(date_end)
+
+    @api.depends(
+        "contract_line_ids.recurring_next_date", "contract_line_ids.is_canceled",
+    )
+    def _compute_recurring_next_date(self):
+        for contract in self:
+            recurring_next_date = contract.contract_line_ids.filtered(
+                lambda l: (
+                    l.recurring_next_date
+                    and not l.is_canceled
+                    and (not l.display_type or l.is_recurring_note)
+                )
+            ).mapped("recurring_next_date")
+            if recurring_next_date:
+                contract.recurring_next_date = min(recurring_next_date)
+            else:
+                contract.recurring_next_date = False
+
+    def _inverse_recurring_next_date(self):
+        for record in self:
+            record.contract_line_ids.recurring_next_date = record.recurring_next_date
```

Comment:
Perhaps this change?
```diff
diff --git a/contract/models/contract.py b/contract/models/contract.py
index 090ce7fe..055ae0fe 100644
--- a/contract/models/contract.py
+++ b/contract/models/contract.py
@@ -212,7 +212,7 @@ class ContractContract(models.Model):
                 contract.recurring_next_date = False
 
     def _inverse_recurring_next_date(self):
-        for record in self:
+        for record in self.filtered(lambda x: not x.line_recurrency):
             record.contract_line_ids.recurring_next_date = record.recurring_next_date
 
     @api.depends("contract_line_ids.create_invoice_visibility")
```

Without this I cannot create contract:
- "Recurrency at line level? = Yes"
- One product line with filled "Date of Next Invoice" (for example 07/22/2020)

When I press "Save" on contract, it fails with message "You must supply a date of next invoice for contract line 'xxxx'" which is not true.